bricks/model/hooks/post_gen.dart

```diff
@@ -0,0 +1,95 @@
+import 'dart:io';
+
+import 'package:mason/mason.dart';
+
+void run(HookContext context) async {
+  final useEquatable = context.vars["use_equatable"];
+  final usesJson = context.vars["use_json"];
+  final modelGenerator =
+      context.vars["model_generator_mode"].toString().toLowerCase();
+  final usesFreezed = modelGenerator == 'freezed';
+
+  final directory = Directory.current.path;
+  final folders = directory.toString().split('/').toList();
+  final index = folders.indexWhere((folder) => folder == 'lib');
+  final root = folders.sublist(0, index).join('/').toString();
+
+  if (useEquatable) {
+    final addEquatableProcess = await Process.start(
+      'flutter',
+      ['pub', 'add', 'equatable'],
+      runInShell: true,
+      workingDirectory: root,
+    );
+    await stdout.addStream(addEquatableProcess.stdout);
+    await stderr.addStream(addEquatableProcess.stderr);
+  }
+
+  if (usesJson) {
+    final addJsonProcess = await Process.start(
+      'flutter',
+      ['pub', 'add', 'json_annotation'],
+      runInShell: true,
+      workingDirectory: root,
+    );
+    await stdout.addStream(addJsonProcess.stdout);
+    await stderr.addStream(addJsonProcess.stderr);
+    final addJsonDevProcess = await Process.start(
+      'flutter',
+      ['pub', 'add', '--dev', 'json_serializable'],
+      runInShell: true,
+      workingDirectory: root,
+    );
+    await stdout.addStream(addJsonDevProcess.stdout);
+    await stderr.addStream(addJsonDevProcess.stderr);
+  }
+
+  if (usesFreezed) {
+    final addFreezedProcess = await Process.start(
+      'flutter',
+      ['pub', 'add', 'freezed_annotation'],
+      runInShell: true,
+      workingDirectory: root,
+    );
+    await stdout.addStream(addFreezedProcess.stdout);
+    await stderr.addStream(addFreezedProcess.stderr);
+    final addBuildRunnerProcess = await Process.start(
+      'flutter',
+      ['pub', 'add', '--dev', 'build_runner'],
+      runInShell: true,
+      workingDirectory: root,
+    );
+    await stdout.addStream(addBuildRunnerProcess.stdout);
+    await stderr.addStream(addBuildRunnerProcess.stderr);
+    final addFreezedDevProcess = await Process.start(
+      'flutter',
+      ['pub', 'add', '--dev', 'freezed'],
+      runInShell: true,
+      workingDirectory: root,
+    );
+    await stdout.addStream(addFreezedDevProcess.stdout);
+    await stderr.addStream(addFreezedDevProcess.stderr);
+  }
+
+  if (useEquatable || usesJson || usesFreezed) {
+    final pubGetProcess = await Process.start(
+      'flutter',
+      ['pub', 'get'],
+      runInShell: true,
+      workingDirectory: root,
+    );
+    await stdout.addStream(pubGetProcess.stdout);
+    await stderr.addStream(pubGetProcess.stderr);
+  }
+
+  if (usesFreezed) {
+    final buildRunnerProcess = await Process.start(
```

Comment:
Ideally, I would not want to run the build runner every time a developer uses the model brick.

It would be good if the generated files were already created (Like how the from/toJson is already created).

Thanks a bunch for looking into this though! @LosDanieloss

Reply:
Thanks for taking a look. 
I think I'm missing something here. Right now if model brick with freezed is used brick create empty `*.g.dart`. Which is then populated by `json_serializable` when `build_runner` runs. Are we talking about creating empty `*.freezed.dart` file and removing build runner invocation? In that case using brick will create code that contains errors. 
On the other hand to remove call to `build_runner` and provide error-free code after brick is done I can only see one scenario in which brick duplicates freezed & json_serializable code generation and create code that looks same as the one created by those generators when `build_runner` ends. However this don't sounds good to me, since changed in any of those two could make brick unusable.
So what I'm missing? I'm new to mason, so maybe there is some online resource that would help me what should I do.

Reply:
All good!

So you pretty much mentioned what we need to do. We need to remove the build runner invocation, and then create the initial generated file ourselves.

This would allow the developer to use the model straight out the box without having to run any generation.

There are multiple ways we could do this, but I think the easiest one is wrapping the files and a conditional (For freezed, json, and none).

Let me know what you think!

Reply:
Understood, will work on it.

Reply:
Hi @LukeMoody01 

it's been a while. I looked into how to implement changes we discussed above and although using brick code to generate `*.freezed.dart` file that mimics the one being generated when running `build_runner` looks straightforward then doing same for `json_serializable` is not trivial task because of generic types. Due to that I don't see the point of doing so. I'm here to discuse further actions. 
I see three possibilities:
- leave things as they're currently in the pull request with usage of `build_runner`
- remove invokation of `build_runner` and leave decision to run it for end user or to the fact that user is using watcher
- someone else will picking up pull request and implemente furher changes needed for model brick to mimic json_serializable & freezed.

Reply:
Hey @LosDanieloss 👋 

I will happily pick these changes up as I feel it is worth it for the developer not to run generation, thanks for all the work you did up till this point!